apps/s_client.c

```diff
@@ -798,6 +801,31 @@ static int new_session_cb(SSL *S, SSL_SESSION *sess)
     return 0;
 }
 
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    BIO *keylog = BIO_new_file(keylog_file, "a");
+    char buffer[256];
```

Comment:
Why the temporary buffer? Why not just pass the line directly to the BIO, and then do a separate write of the new line character?